cc3d/core/MaBoSSCC3D.py

```diff
@@ -16,30 +16,48 @@
 """
 import os
 import tempfile
-from typing import Dict
+from typing import Dict, Optional
 from cc3d.cpp import CompuCell
+from cc3d.core.CellInventoryWatcher import CellInventoryWatcher
 try:
     from cc3dext import MaBoSSCC3DPy
     maboss_engine_type = MaBoSSCC3DPy.CC3DMaBoSSEngine
+    maboss_engine_container_type = MaBoSSCC3DPy.CC3DMaBoSSEngineContainer
     __has_extension__ = True
 except ModuleNotFoundError:
     __has_extension__ = False
     maboss_engine_type = object
+    maboss_engine_container_type = object
 
 
 class NoMaBoSSExtensionError(Exception):
     """Simple exception to notify of missing required extension"""
     pass
 
 
+class MaBoSSInventoryWatcher(CellInventoryWatcher):
+    """
+    Inventory watcher that maintains synchronization between the cell and engine inventories
+    """
+
+    def on_cell_remove(self, cell: CompuCell.CellG):
+        from cc3d.CompuCellSetup import persistent_globals as pg
```

Comment:
can we move this import to the top of the file?

Reply:
Done